api/ccapi/include/layer.h

```diff
@@ -259,6 +259,10 @@ class Layer {
    * @note      layer needs to be finalized before called.
    */
   virtual void setWeights(const std::vector<float *>) = 0;
+
+  virtual void save_kvcache(){};
+
+  virtual void load_kvcache(){};
```

Comment:
I also think similarly to @Eunju. It seems reasonable to create a new API only when it is necessary in all cases.
Instead of creating KV_Cache Load and Save, how about passing it as a parameter in the Load function?